src/braket/default_simulator/openqasm/circuit.py

```diff
@@ -60,6 +61,12 @@ def add_instruction(self, instruction: [GateOperation, KrausOperation]) -> None:
         self.instructions.append(instruction)
         self.qubit_set |= set(instruction.targets)
 
+    def add_measure(self, target: tuple[int]):
+        for qubit in target:
+            if qubit in self.measured_qubits:
+                raise ValueError(f"Qubit {qubit} is already measured or captured.")
+        self.measured_qubits += [qubit for qubit in target]
```

Comment:
Nit: this would be slightly more performant if you used a dict, which is guaranteed to maintain order of keys since python 3.6. Number of qubits is pretty small, so I think this is alright for now, but also wouldn't mind updating. Your call.

If you choose to keep as is, I'd leave a comment that it can be improved in the future.

Reply:
Just to clarify, do you mean use a dict for `self.measured_qubits` here?

Reply:
Per our offline discussion, we'll leave it as a list, since the number of qubits is small and in the future we may want to have duplicates

Reply:
We could append the qubits to measured_qubits during the loop to avoid iterating through target twice

Reply:
Great idea, I'll move that inside the loop